### PR TITLE
Fix deprecation warning on secure_json_data

### LINKED ISSUE #626: Confusing textual error in Arbitrary JSON Data message
### Terraform Version
* Terraform: latest
* Terraform Grafana Provider: Latest
* Grafana: 8.5.10.

### Affected Resource(s)
* grafana_data_source

### Terraform Configuration Files
```hcl
resource "grafana_data_source" "my_data" {
  name          = "xxx"
  type          = "mysql"
  url           = xxx
  username      = "xxx"
  database_name = "xxx"
  is_default    = false

  secure_json_data {
    password = data.aws_ssm_parameter.password.value
  }
```

### Expected Behavior
Correct deprecation message that `secure_json_data` is deprecated. See https://github.com/grafana/terraform-provider-grafana/pull/607

### Actual Behavior
Confusing errormessage:

```
Warning: Argument is deprecated
with module.xxx.grafana_data_source.xxx
on ../../modules/xxx/grafana.tf line xxx, in resource "grafana_data_source" "xxx":
resource "grafana_data_source" "xxx" {
Use secure_json_data instead. It supports arbitrary JSON data, and therefore all attributes.
```

Show be `secure_json_data_map ` according to https://github.com/grafana/terraform-provider-grafana/pull/607

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- https://github.com/grafana/terraform-provider-grafana/pull/607)

### PR DESCRIPTION
**Pull request by fatbasstard**
Fixes: https://github.com/grafana/terraform-provider-grafana/issues/626

### CODE REVIEW

File: grafana/resource_data_source.go [hunk 1]
```diff
@@ -461,8 +461,8 @@ source selected (via the 'type' argument).
 				Type:        schema.TypeList,
 				Optional:    true,
 				Sensitive:   true,
-				Description: "Deprecated: Use secure_json_data instead. It supports arbitrary JSON data, and therefore all attributes.",
-				Deprecated:  "Use secure_json_data instead. It supports arbitrary JSON data, and therefore all attributes.",
+				Description: "Deprecated: Use secure_json_data_map instead. It supports arbitrary JSON data, and therefore all attributes.",
+				Deprecated:  "Use secure_json_data_map instead. It supports arbitrary JSON data, and therefore all attributes.",
```
Comment (julienduchesne) [inline grafana/resource_data_source.go]:
```suggestion
				Description: "Deprecated: Use secure_json_data_encoded instead. It supports arbitrary JSON data, and therefore all attributes.",
				Deprecated:  "Use secure_json_data_encoded instead. It supports arbitrary JSON data, and therefore all attributes.",
```

Comment (fatbasstard) [reply grafana/resource_data_source.go]: Just checked the [documentation](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/data_source#secure_json_data_encoded) (should have done that earlier I guess, sorry. Will update the PR.

**Comment by fatbasstard** [comment]
What about `password` and `basic_auth_password` btw? That mentions `Use secure_json_data.basic_auth_password instead. This attribute is removed in Grafana 9.0+.` Should that be updated to point to `secure_json_data_encoded` as well?

**Comment by julienduchesne** [comment]
> What about `password` and `basic_auth_password` btw? That mentions `Use secure_json_data.basic_auth_password instead. This attribute is removed in Grafana 9.0+.` Should that be updated to point to `secure_json_data_encoded` as well?

Oh, yeah. I missed the `password` attributes, they should point to the new attribute, indeed

**Comment by julienduchesne** [comment]
Thanks. LGTM. Can you run `go generate`?

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**